app/models/medium.rb

```diff
@@ -295,11 +304,12 @@ def self.search_by(search_params, page)
     if search_params[:all_tags] == '1' && search_params[:tag_operator] == 'and'
       search_params[:tag_ids] = Tag.pluck(:id)
     end
-    admin = User.find_by(id: search_params[:user_id])&.admin?
+    user = User.find_by(id: search_params[:user_id])
     search = Sunspot.new_search(Medium)
     search.build do
       with(:sort, search_params[:types])
-      without(:sort, Medium.advanced_sorts) unless admin
+      without(:sort, 'RandomQuiz')
+      without(:sort, Medium.advanced_sorts) unless user&.admin_or_editor?
```

Comment:
That's why the "hidden" field in the HTML is no security issue since we filter "advanced types" out here, right?

Reply:
Yes, even if you were to manipulate the form to set `all_types` to '1', you would not get these "advanced types" because of this line(now that I write, I indeed note that type and sort are pretty mixed throughout this code).